hibernate-core/src/main/java/org/hibernate/dialect/PostgreSQLDialect.java

```diff
@@ -452,10 +452,10 @@ public void initializeFunctionRegistry(QueryEngine queryEngine) {
 		queryEngine.getSqmFunctionRegistry().registerBinaryTernaryPattern(
 				"locate",
 				queryEngine.getTypeConfiguration().getBasicTypeRegistry().resolve( StandardBasicTypes.INTEGER ),
-				"position(?2 in ?1)",
-				"(position(?2 in substring(?1 from ?3))+(?3)-1)",
+				"position(?1 in ?2)",
+				"(position(?1 in substring(?2 from ?3))+(?3)-1)",
 				FunctionParameterType.STRING, FunctionParameterType.STRING, FunctionParameterType.INTEGER
-		).setArgumentListSignature("(string, pattern[, start])");
+		).setArgumentListSignature("(pattern, string[, start])");
```

Comment:
I think this one here should use `CommonFunctionFactory.locate_positionSubstring` as well

Reply:
I just tried this out on PostgreSQL and the pattern `(position(?1 in substring(?2 from ?3))+(?3)-1)` is correct, so I actually think that `CommonFunctionFactory.locate_positionSubstring` might be wrong because it misses the `-1` at the end.